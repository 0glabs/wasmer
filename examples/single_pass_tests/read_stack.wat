(module
    (type $t1 (func))
	(func $stack_read (import "wasi_unstable" "stack_read") (type $t1))

    (func $_start (export "_start")
        (local $x i32)
        (i32.const 1)
        (i32.const 2)
        (i32.add)

        (i32.const 1)

        (i32.const 3)
        (i32.const 4)
        (i32.add)

        (call $f1
            (i32.const 10)
            (i32.const 42)
            (i32.const 43)
            (i32.const 44)
            (i32.const 45)
            (i32.const 46)
            (i32.const 47)
            (i32.const 48)
            (i32.const 49)
        )

        (drop)
        (drop)
        (drop)
    )

    (func $f1 (param $x i32) (param $v1 i32) (param $v2 i32) (param $v3 i32) (param $v4 i32) (param $v5 i32) (param $v6 i32) (param $v7 i32) (param $v8 i32)
        (if (i32.eq (get_local $x) (i32.const 1))
            (then call $stack_read)
            (else (call $f2
                (i32.sub (get_local $x) (i32.const 1))
                (get_local $v1)
                (get_local $v2)
                (get_local $v3)
                (get_local $v4)
                (get_local $v5)
                (get_local $v6)
                (get_local $v7)
                (get_local $v8)
            ))
        )
    )

    (func $f2 (param $x i32) (param $v1 i32) (param $v2 i32) (param $v3 i32) (param $v4 i32) (param $v5 i32) (param $v6 i32) (param $v7 i32) (param $v8 i32)
        (if (i32.eq (get_local $x) (i32.const 1))
            (then call $stack_read)
            (else (call $f1
                (i32.sub (get_local $x) (i32.const 1))
                (get_local $v1)
                (get_local $v2)
                (get_local $v3)
                (get_local $v4)
                (get_local $v5)
                (get_local $v6)
                (get_local $v7)
                (get_local $v8)
            ))
        )
    )
)
