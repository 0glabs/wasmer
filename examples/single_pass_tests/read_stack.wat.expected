Frame #0: 0x110f25251 WasmFunctionState { stack: [], locals: [1, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #1: 0x110f25195 WasmFunctionState { stack: [], locals: [2, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #2: 0x110f2529c WasmFunctionState { stack: [], locals: [3, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #3: 0x110f25195 WasmFunctionState { stack: [], locals: [4, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #4: 0x110f2529c WasmFunctionState { stack: [], locals: [5, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #5: 0x110f25195 WasmFunctionState { stack: [], locals: [6, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #6: 0x110f2529c WasmFunctionState { stack: [], locals: [7, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #7: 0x110f25195 WasmFunctionState { stack: [], locals: [8, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #8: 0x110f2529c WasmFunctionState { stack: [], locals: [9, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #9: 0x110f25195 WasmFunctionState { stack: [], locals: [10, 42, 43, 44, 45, 46, 47, 48, 49] }
Frame #10: 0x110f25090 WasmFunctionState { stack: [Some(3), None, Some(7)], locals: [0] }
